docs/scos/dev/front-end-development/202212.0/oryx/oryx-server-side-rendering.md

```diff
@@ -0,0 +1,122 @@
+---
+
+Server-side rendering (SSR), including Static Site Generation (SSG) as a variant, has grown in popularity due to its ability to boost web application performance, facilitate effective Search Engine Optimization (SEO), social sharing, and improve Core Web Vitals (CWV). By delivering pre-rendered HTML from the server or even a  content delivery network (CDN) to the client, SSR and SSG lead to quicker initial page load times, improve user experience (UX), and can significantly improve CWV scores. SSG, in particular, pre-renders HTML at build time, resulting in static HTML, CSS, and JavaScript files that can be served directly from a CDN. It is a useful strategy for sites with content that does not change frequently, and can improve performance, scalability, and security.
+
+## Differences between server-side and client-side rendering
+
+* Speed and performance: With server-side rendering (SSR), the browser receives pre-rendered HTML, reducing the time taken to display meaningful content.
+
+* SEO: SSR is typically more favorable for SEO because search engine crawlers find it easier to index pre-rendered HTML content.
+
+* Social media integration: SSR significantly improves integration with social providers like Facebook and Twitter and bots like Slack. It enables the generation of link previews, rich snippets, and thumbnails, enhancing the visibility and appeal of shared content on these platforms.
+
+* User experience (UX): By delivering pre-rendered content faster, SSR minimizes user waiting time, providing a superior user experience compared to client-side rendering (CSR).
+
+* Resource allocation: While SSR enhances performance and user experience, it requires more server resources and processing power. CSR lightens the server load by offloading rendering to the client but at the cost of potentially increased load times and less effective SEO.
+
+
+## Advantages and disadvantages of SSR
+
+| ADVANTAGES OF SSR                        | DISADVANTAGES of SSR    |
+|-------------------------------------------------------------------|---------------------|
+| Quicker initial page load time.                                  | Higher server resource usage. |
+| Enhanced SEO.                                                     | Higher development and deployment complexity. |
+| Improved UX.                                                      | Infrastructure concerns. |
+| Better social media integration.                                  | Potential for stale content. |
+
+## Applicability of SSR
+
+While SSR offers numerous benefits, it's not the best fit for every type of application, like the following:
+
+- B2B shops with restricted access: public-facing SEO and quicker initial page load times offered by SSR may not significantly benefit these types of applications.
+- Business apps: applications heavily focused on functionality, like a fulfillment app, might not require the SEO or user experience enhancements provided by SSR.
```

Comment:
Business apps: applications heavily focused on functionality, such as a fulfilment app, might not require the SEO or user experience enhancements provided by SSR.